### PR TITLE
Assertion for TextInputLayout Text Helper Hint

### PR DESCRIPTION
**Pull request by IanArb**
This commit will add new assertion for TextInputLayout's helper text 

- Created feature request: https://github.com/SchibstedSpain/Barista/issues/302
- Implemented a new class for asserting the helper text from the TextInputLayout
- Added Android tests to demonstrate the behaviour
- Added example to an existing layout file activity_hintanderrortext.xml
- Updated README for documentation for implementation details

### CODE REVIEW

**Comment by IanArb** [comment]
@Sloy Hey there it seems I can't reviewers to this PR :(

**Comment by alorma** [comment]
Hi @IanArb 

I've added @SchibstedSpain/barista-maintainers team to review, so we will review it.

Thanks for your contribution

File: library/src/main/java/com/schibsted/spain/barista/assertion/BaristaHelperTextAssertions.kt [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    val resourceString = InstrumentationRegistry.getTargetContext().resources.getString(text)
+    assertHelperText(viewId, resourceString)
+  }
+
+  @JvmStatic
+  fun assertHelperText(@IdRes viewId: Int, text: String) {
+    ViewMatchers.withId(viewId).assertAny(matchError(text))
+  }
+
+  private fun matchError(expectedHelperText: String): Matcher<View> {
```
Comment (alorma) [inline library/src/main/java/com/schibsted/spain/barista/assertion/BaristaHelperTextAssertions.kt]: Is there a way to differentiate between helper and error text?

Comment (IanArb) [reply library/src/main/java/com/schibsted/spain/barista/assertion/BaristaHelperTextAssertions.kt]: Doh! Just took the error sample and forgot to rename the method. Will change to `matchAssistiveHelperText`

**Comment by IanArb** [comment]
> Captura de pantalla 2019-05-31 a las 9 34 15
> 
> Hi @IanArb
> 
> As reading on Material.io guidelines, this helper and error texts are `assistive elements`, I would rename the methods to something like `assertAssistiveText()` what do you think?

Yep that sounds perfectly reasonable to me. Would be in favour of renaming the object class name as well.

**Comment by IanArb** [comment]
Hey @alorma created a new commit with the suggested changes.

**Comment by alorma** [comment]
@Sloy !

**Comment by alorma** [comment]
Good work, merging it

**Comment by alorma** [comment]
Available at release 3.1.0!
https://github.com/SchibstedSpain/Barista/releases/tag/v3.1.0

**Pull request merged**